src/qmapshack/gis/trk/CToRouteNameProjectDialog.cpp

```diff
@@ -0,0 +1,295 @@
+/**********************************************************************************************
+    Copyright (C) 2021 Oliver Eichler <oliver.eichler@gmx.de>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+**********************************************************************************************/
+
+#include "canvas/CCanvas.h"
+#include "CMainWindow.h"
+#include "gis/CGisDatabase.h"
+#include "gis/CGisListWks.h"
+#include "gis/db/CDBProject.h"
+#include "gis/db/CSelectDBFolder.h"
+#include "gis/db/CSetupFolder.h"
+#include "gis/gpx/CGpxProject.h"
+#include "gis/qms/CQmsProject.h"
+#include "helpers/CSettings.h"
+#include "CToRouteNameProjectDialog.h"
+
+#include <QtWidgets>
+
+CToRouteNameProjectDialog::CToRouteNameProjectDialog(QString& projectName, QString& routeName,
+                                                     IGisProject::type_e& type)
+    : QDialog(CMainWindow::getBestWidgetForParent())
+    , routeName(routeName)
+    , projectName(projectName)
+    , type(type)
+{
+    setupUi(this);
+    SETTINGS;
+    cfg.beginGroup("RouteToTrack");
+    saveSubPoints = cfg.value("saveSubPoints", false).toBool();
+    cfg.endGroup();
+
+    checkBoxSubPoints->setChecked(saveSubPoints);
+    key = IGisProject::getUserFocus();
+    QListWidgetItem* lastSelectedItem = nullptr;
+    const QTreeWidget *parent = CGisWorkspace::self().getWks();
```

Comment:
Ok, I see what's happening here. You are duplicating code from CGisWorkspace::selectProject().  Please use as much of the existing objects and dialogs as possible. Duplicating code drives every project into a mess that is hard to maintain.

Reply:
This would require a more comprehensive redesign of CGisWorkspace::selectProject(). I am not sure how to implement this. But at least the function getNameAndProject would benefit, since the name and project selection could be integrated into one dialog instead of two.

Do you have an Idea or some tips?

Reply:
The idea is to have something like that
![image](https://user-images.githubusercontent.com/54329441/120919334-54452880-c6b9-11eb-915a-542d1f3d6576.png)

The project is initially the project of the track. With the tool button on the right you open the usual  CGisWorkspace::selectProject() dialog. The rest is pre-set route name and other options like selecting subpoints or some future number of route points limit.

Reply:
Pushed a new commit with the modified dialog.